br/pkg/restore/data.go

```diff
@@ -332,41 +353,60 @@ type Regions struct {
 // generate the related the recovery plan to tikvs:
 // 1. check overlap the region, make a recovery decision
 // 2. build a leader list for all region during the tikv startup
-// 3. TODO: set region as tombstone
-// TODO: This function has to refactoring, leader selection shall follow the design
-// this is temp solution, peer with the max last index will be a leader.
+// 3. get max allocate id
 func (recovery *Recovery) makeRecoveryPlan() {
-	type peer struct {
+	type Peer struct {
 		rid uint64
 		ver uint64
 	}
 
-	type regionEndKey struct {
+	type RegionEndKey struct {
```

Comment:
seems this struct can be replaced by Regions(rename to Peer) too, just add another `prefixed end key` field